### ISSUE TITLE
RHF and ROHF stability analysis = follow not implemented

### ISSUE THREAD

**Issue by susilehtola** [psi4/psi4#816]
Looks like psi4 is not following instabilities with ROHF. The issue is present in the current master.

Debug case:
```
molecule mol {
0 1
H 0.530330 0.000000 0.0
H 0.000000 0.530330 0.0
H -0.530330 0.000000 0.0
H -0.000000 -0.530330 0.0
}

memory 31500 mb

set basis aug-pcseg-1
set reference rohf
set guess core
set scf_type pk
set stability_analysis follow
scf_e, scf_wfn = energy('scf', return_wfn = True)
```

This converges as
```
   @ROHF iter   1:    -1.35786249373809   -1.35786e+00   3.29066e-02 
   @ROHF iter   2:    -1.76576328233758   -4.07901e-01   2.42353e-03 DIIS
   @ROHF iter   3:    -1.76782078804299   -2.05751e-03   6.21952e-04 DIIS
   @ROHF iter   4:    -1.76802124530854   -2.00457e-04   4.60781e-05 DIIS
   @ROHF iter   5:    -1.76802266421582   -1.41891e-06   3.48831e-06 DIIS
   @ROHF iter   6:    -1.76802267513349   -1.09177e-08   6.89002e-07 DIIS
    SO Integrals not on disk, computing...done.
        Presorting SO-basis two-electron integrals.
        Sorting File: SO Ints (nn|nn) nbuckets = 1
        Transforming the one-electron integrals and constructing Fock matrices
        Starting first half-transformation.
        Sorting half-transformed integrals.
        First half integral transformation complete.
        Starting second half-transformation.
        Two-electron integral transformation complete.
        Starting first half-transformation.
        Sorting half-transformed integrals.
        First half integral transformation complete.
        Starting second half-transformation.
        Two-electron integral transformation complete.
    Lowest ROHF->ROHF stability eigenvalues:-
     B1g -0.022531      B2u 0.142418       B3g 0.159990       B1g 0.196160  
```
The lowest eigenvalue is significantly negative, and the solution is indeed a saddle point. However, the program does **not** proceed by a SCF restart from displaced orbitals.

**Comment by susilehtola** [comment]
Indeed, it looks like ```FOLLOW``` has only been implemented in ```libscf_solver/uhf.cc```, and it's missing both from ```libscf_solver/rhf.cc``` and ```libscf_solver/rohf.cc```...!

**Comment by lothian** [comment]
Hmmm...I don't think the code can't follow an instability that breaks the symmetry of the orbitals.  Have you tried lowering the symmetry such that B1g becomes the totally symmetric irrep?

**Comment by susilehtola** [comment]
It finds the true ground state in C1, but gets the saddle point in any other symmetry.

**Comment by lothian** [comment]
OK, so you can get around the issue, but my point is that even if the follow algorithm were implemented, you'd need to drop the symmetry before the code could track the instability.

**Comment by susilehtola** [comment]
Doesn't the code allow symmetry broken orbitals?

**Comment by lothian** [comment]
To my knowledge, the code doesn't have the ability to change symmetries in the middle of the calculation.

**Comment by jgonthier** [comment]
I did the implementation of FOLLOW for uhf ( and modernized the code) but I didn't have time to do rhf or rohf. I thought somebody had gotten to modernize these two as well, though. In any case, I think lothian is right: the stability analysis only looks for minima in the same symmetry.

**Comment by dgasmith** [comment]
Noted in #886 and #1005. Should be implemented in Psi 1.3.
